public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -80,9 +80,11 @@ export const DataConnection = (props: any) => {
       disabled: false,
       content: (
         <AccessControlTab
+          allowedRoles={datasourceDetails.allowedRoles}
           dataConnection={dataSource}
           connector={datasourceDetails.connector}
           properties={datasourceDetails.properties}
+          key={JSON.stringify(datasourceDetails.allowedRoles)}
```

Comment:
Needed to cause re-render when allowedRoles changes

Reply:
if allowedRoles changes, current component state changes and `tabs` variable is re-created, why does it not rerender?

Reply:
I am not sure. This is just what I saw while testing it locally. It might be because the Tabs component is doing some kind of memoization under the hood on the in-focus tab, but adding this in fixed the problem and I don't think it could make anything worse?